packages/elasticsearch-asset-apis/src/elasticsearch-reader-api/ElasticsearchReaderAPI.ts

```diff
@@ -111,21 +112,100 @@ export class ElasticsearchReaderAPI {
     /**
      * Fetch a given slice, the data will be returned in the format
      * specified under `response_type` in the config
-    */
+     *
+     * `fetch` now sets the query `size` parameter to 1.5 times the count
+     * recorded in the slice, rather than the window size of the index.
+     * It will retry up to five times if the size of the result set comes back
+     * the same size as the querySize, this is meant to ensure all of the
+     * records are retrieved, but should be unlikely to happen in most cases
+     *
+     * When it previously used the indices max window size, it appeared to
+     * slow ES queries down significantly, refs:
+     *   https://github.com/terascope/elasticsearch-assets/issues/948
+     */
     async fetch(queryParams: ReaderSlice = {}): Promise<DataEntity[]|DataFrame|Buffer> {
-        // attempt to get window if not set
+        const countExpansionFactor = 1.5;
+        let querySize = 10000;
+        const retryLimit = 5;
+
+        // attempt to get window if not set (sets this.windowSize as side effect)
         if (!this.windowSize) await this.setWindowSize();
 
-        // if we did go ahead and complete query
-        const query = buildQuery(this.config, {
-            ...queryParams, count: this.windowSize
-        });
+        // set querySize to the lesser of expandedSize or windowSize
+        if (queryParams.count) {
+            const expandedSize = Math.ceil(queryParams.count * countExpansionFactor);
+            if (this.windowSize) {
+                if (expandedSize >= this.windowSize) {
+                    throw new Error(`The query size, ${expandedSize}, is greater than the index.max_result_window: ${this.windowSize}`);
```

Comment:
NOTE: I treat this as a fault because if the number of records in a slice exceeds the max_window_size, there's no way to get them out.  This isn't likely to happen except in cases where max_window_size is set too low.